internal/dataplane/kong_client.go

```diff
@@ -483,17 +487,16 @@ func (c *KongClient) Update(ctx context.Context) error {
 
 	const isFallback = false
 	shas, gatewaysSyncErr := c.sendOutToGatewayClients(ctx, parsingResult.KongState, c.kongConfig, isFallback)
-	konnectSyncErr := c.maybeSendOutToKonnectClient(ctx, parsingResult.KongState, c.kongConfig, isFallback)
+
+	// TODO: only send Kong configuration to Konnect when applied configuration to gateways successfully(`gatewaysyncErr != nil`)?
```

Comment:
IMHO we should not send the broken config to Konnect as that would return errors at some point anyway. Instead we should issue error logs and consider adding this information into the events that are generated.

Reply:
Done.